test/github-issues/10412/issue-10412.ts

```diff
@@ -0,0 +1,51 @@
+import "reflect-metadata"
+import {
+    createTestingConnections,
+    closeTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { DataSource } from "../../../src/index.js"
+import { expect } from "chai"
+import { documentRelationEntitySchema } from "./entity/entities"
+
+describe("github issues > #10412 EntityPropertyNotFoundError when joinTable is used", () => {
```

Comment:
We are moving away from creating tests for specific GitHub issues. Instead we need tests which demonstrate that `EntityMetadataBuilder` works correctly. Unfortunately there isn't a test suite for this class yet, so one needs to be created.

Please move this test file to `test/functional/entity-metadata/entity-metadata-builder.ts` and adjust test descriptions appropriately for this context.